tempesta_fw/http.c

```diff
@@ -2303,8 +2302,13 @@ tfw_http_req_process(TfwConn *conn, struct sk_buff *skb, unsigned int off)
 		 * released when handled in tfw_cache_req_process() below.
 		 * So, save the needed request flag for later use as it
 		 * may not be accessible later through @req->flags.
+		 * If the connection must be closed, it also should be marked
+		 * with @Conn_Stop flag - to left it alive for sending responses
+		 * and, at the same time, to stop passing data for processing
+		 * from the lower layer.
 		 */
-		req_conn_close = req->flags & TFW_HTTP_CONN_CLOSE;
+		if((req_conn_close = req->flags & TFW_HTTP_CONN_CLOSE))
+			TFW_CONN_TYPE(req->conn) |= Conn_Stop;
```

Comment:
It has sense to remove `req_conn_close` variable and just use `TFW_CONN_TYPE(req->conn) | Conn_Stop` check instead. That simplifies the code for one variable.

Reply:
The request may be released in `tfw_http_req_cache_cb` (if it will be served from cache) - so, connection descriptor can be inaccessible through `req->conn` after `tfw_cache_process`.